varats/utils/cli_util.py

```diff
@@ -79,6 +82,29 @@ def initialize_logger_config() -> None:
     logging.basicConfig(level=log_level)
 
 
+ConcreteResearchTool = tp.TypeVar(
+    "ConcreteResearchTool", bound=ResearchTool[SpecificCodeBase]
+)
+
+
+def get_research_tool_type(name: str) -> tp.Type[ConcreteResearchTool]:
```

Comment:
According to https://github.com/python/mypy/issues/2756, this `TypeVar`s cannot depend on `TypeVar`s in the current type system of python. Replacing `SpecificCodeBase` with `CodeBase` or `Any` did not help either, as the types of the concrete research tools cannot be matched then.

The best solution I could find is:
```suggestion
def get_research_tool_type(name: str) -> tp.Type[ResearchTool[tp.Any]]:
```